src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -1058,105 +1098,50 @@ where
     }
 
     async fn flush(&mut self) -> io::Result<()> {
-        self.stream
-            .as_mut()
-            .unwrap()
-            .write_all(&self.write_buf)
-            .await?;
+        let mut stream = self.stream.lock().await;
+        match &mut *stream {
+            PgStreamInner::Placeholder => unreachable!(),
+            PgStreamInner::Unencrypted(stream) => {
+                stream.write_all(&self.write_buf).await?;
+                stream.flush().await?;
+            }
+            PgStreamInner::Ssl(ssl_stream) => {
+                ssl_stream.write_all(&self.write_buf).await?;
+                ssl_stream.flush().await?;
+            }
+        }
         self.write_buf.clear();
-        self.stream.as_mut().unwrap().flush().await?;
         Ok(())
     }
-
-    fn stream(&mut self) -> &mut (impl AsyncRead + Unpin + AsyncWrite) {
-        self.stream.as_mut().unwrap()
-    }
-}
-
-/// The logic of Conn is very simple, just a static dispatcher for TcpStream: Unencrypted or Ssl:
-/// Encrypted.
-pub enum Conn<S> {
```

Comment:
`Conn` is removed because the variants are now under the `Arc` in `PgStream`, so that we can upgrading the underlying stream for all shared `PgStream` instances.